src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -171,20 +172,55 @@ else if ( clazz.equals(int[].class) ) {
     public static List<Allele> calculateMostLikelyAlleles(final VariantContext vc, final int defaultPloidy,
                                                           final int numAltAllelesToKeep) {
         Utils.nonNull(vc, "vc is null");
-        final int nonRefAltAlleleIndex = GATKVariantContextUtils.indexOfAltAllele(vc, GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE, false);
-        final int[] properAltAlleleIndices = IntStream.range(1, vc.getNAlleles()).filter(n -> n != nonRefAltAlleleIndex).toArray();
+        Utils.validateArg(defaultPloidy > 0, () -> "default ploidy must be > 0 but defaultPloidy=" + defaultPloidy);
+        Utils.validateArg(numAltAllelesToKeep > 0, () -> "numAltAllelesToKeep must be > 0, but numAltAllelesToKeep=" + numAltAllelesToKeep);
 
-        if (numAltAllelesToKeep >= properAltAlleleIndices.length) {
+        final boolean hasSymbolicNonRef = vc.hasAllele(GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE);
+        final int numberOfProperAltAlleles = hasSymbolicNonRef ? vc.getNAlleles() - 1 : vc.getNAlleles() - 2; // 1 for reference, 2 for references and NON_REF_SYMBOLIC_ALLELE
+
+        if (numAltAllelesToKeep >= numberOfProperAltAlleles) {
             return vc.getAlleles();
         }
 
+        final double[] likelihoodSums = calculateLikelihoodSums(vc, defaultPloidy);
+        return filterToMaxNumberOfAltAllelesBasedOnScores(numAltAllelesToKeep, vc.getAlleles(), likelihoodSums);
+    }
+
+
+    /**
+     * @return a list of the best proper alt alleles based on the likelihood sums, keeping the reference allele and {@link GATKVCFConstants#NON_REF_SYMBOLIC_ALLELE}
+     * the list will include no more than {@code numAltAllelesToKeep + 2} alleles and will maintain the order of the original alleles in {@code vc}
+     *
+     */
+    @VisibleForTesting
+    static List<Allele> filterToMaxNumberOfAltAllelesBasedOnScores(int numAltAllelesToKeep, List<Allele> alleles, double[] likelihoodSums) {
+        final int nonRefAltAlleleIndex = alleles.indexOf(GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE);
+        final int numAlleles = alleles.size();
+        final Set<Integer> properAltIndexesToKeep = IntStream.range(1, numAlleles).filter(n -> n != nonRefAltAlleleIndex).boxed()
+                                                             .sorted(Comparator.comparingDouble((Integer n) -> likelihoodSums[n]).reversed())
+                                                             .limit(numAltAllelesToKeep)
+                                                             .collect(Collectors.toSet());
+
+        final List<Allele> mostLikelyAlleles = new ArrayList<>();
+        for ( int i = 0; i < numAlleles; i++){
+            if( i == 0 || i == nonRefAltAlleleIndex || properAltIndexesToKeep.contains(i) ){
+                mostLikelyAlleles.add(alleles.get(i));
+            }
+        }
+        return mostLikelyAlleles;
+    }
+
+    private static double[] calculateLikelihoodSums(final VariantContext vc, final int defaultPloidy) {
```

Comment:
You can file the issue and assign me.  I'm pretty confident that it works for the purposes of your PR getting merged because this is old code that I simply moved into `AlleleSubsettingUtils`.